codec/src/main/java/io/netty/handler/codec/json/JsonObjectDecoder.java

```diff
@@ -88,6 +90,11 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
             return;
         }
 
+        if (this.idx > in.readerIndex() && lastReaderIndex != in.readerIndex()) {
+            this.idx = in.readerIndex();
+            reset();
```

Comment:
Yes, what was happening was the bytebuff was getting reset and this.idx was still pointed to somewhere in the middle.